modules/HapticGlove_module/src/RobotMotorsEstimation.cpp

```diff
@@ -0,0 +1,219 @@
+    if (!YarpHelper::getDoubleFromSearchable(config, "samplingTime", dt))
+    {
+        yError() << m_logPrefix << "unable to find the samplingTime";
+        return false;
+    }
+
+    no_states_kf = config.check("no_states_kf", yarp::os::Value(3)).asInt();
+
+    no_measurement_kf = config.check("no_measurement_kf", yarp::os::Value(1)).asInt();
+
+    yarp::sig::Vector Q_vector(no_states_kf, 0.0), R_vector(no_measurement_kf, 0.0);
+    if (!YarpHelper::getYarpVectorFromSearchable(config, "r_matrix_kf", R_vector))
+    {
+        yError() << m_logPrefix << "Initialization failed while reading r_matrix_kf.";
+        return false;
+    }
+
+    if (!YarpHelper::getYarpVectorFromSearchable(config, "q_matrix_kf", Q_vector))
+    {
+        yError() << m_logPrefix << "Initialization failed while reading q_matrix_kf.";
+        return false;
+    }
+
+    Eigen::MatrixXd Q = Eigen::MatrixXd::Identity(no_states_kf, no_states_kf);
+    Eigen::MatrixXd R = Eigen::MatrixXd::Identity(no_measurement_kf, no_measurement_kf);
+    for (int i = 0; i < no_states_kf; i++)
+    {
+        Q(i, i) = Q_vector(i);
+    }
+    for (int i = 0; i < no_measurement_kf; i++)
+    {
+        R(i, i) = R_vector(i);
+    }
+
+    for (int i = 0; i < m_numOfMotors; i++)
+    {
+        Estimator motorEstimator(dt, R, Q);
+        m_motorEstimatorVector.push_back(motorEstimator);
+    }
+
+    m_z = Eigen::MatrixXd::Zero(no_measurement_kf, 1);
+
+    m_x_hat = Eigen::MatrixXd::Zero(no_measurement_kf, 1);
+    m_P.resize(std::pow(no_states_kf, 2));
+
+    return true;
+}
+
+bool Estimators::initialize(const std::vector<double>& z0)
+{
+    for (size_t i = 0; i < m_numOfMotors; i++)
+    {
+        m_z(0, 0) = z0[i];
+        m_motorEstimatorVector[i].initialize(m_z);
+    }
+    m_isInitialized = true;
+    return true;
+}
+
+bool Estimators::estimateNextState(const std::vector<double>& z)
+{
+    for (size_t i = 0; i < m_numOfMotors; i++)
+    {
+        m_z(0, 0) = z[i];
+        m_motorEstimatorVector[i].estimateNextState(m_z);
+    }
+
+    return true;
+}
+
+bool Estimators::estimateNextSteadyState(const std::vector<double>& z)
+{
+
+    for (size_t i = 0; i < m_numOfMotors; i++)
+    {
+        m_z(0, 0) = z[i];
+        m_motorEstimatorVector[i].estimateNextSteadyState(m_z);
+    }
+    return true;
+}
+
+bool Estimators::getInfo(Eigen::VectorXd& estimatedValues,
+                         Eigen::VectorXd& estimatedVelocities,
+                         Eigen::VectorXd& estimatedAccelerations,
+                         Eigen::MatrixXd& P)
+{
+    if (estimatedValues.size() != m_numOfMotors)
+    {
+        estimatedValues.resize(m_numOfMotors, 0.0);
+    }
+
+    if (estimatedVelocities.size() != m_numOfMotors)
+    {
+        estimatedValues.resize(m_numOfMotors, 0.0);
+    }
+
+    if (estimatedAccelerations.size() != m_numOfMotors)
+    {
+        estimatedValues.resize(m_numOfMotors, 0.0);
+    }
+
+    if (P.rows() != m_numOfMotors && P.cols() == std::pow(m_n, 2))
```

Comment:
Nitpick, ``std::pow`` is usually more expensive than simply ``m_n * m_n``. See https://stackoverflow.com/questions/2940367/what-is-more-efficient-using-pow-to-square-or-just-multiply-it-with-itself